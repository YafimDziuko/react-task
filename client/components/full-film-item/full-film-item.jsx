import React from 'react';
import ReactDOM from 'react-dom';

import './full-film-item.scss';
import { Button } from '../button/button';

export class FullFilmItem extends React.Component {
    render() {
        return (
            <div className="full-film-item">
                <img src={"http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg"} />
                <div className="full-film-info">
                    <div className="title-and-rating">
                        <p className="title">{this.props.title}</p>
                        <p className="rating">{this.props.rating}</p>
                    </div>
                    <p>{this.props.additional}</p>
                    <p className="year-and-time">{this.props.year}    {this.props.duration} min</p>
                    <p className="description">{this.props.description}</p>
                    <Button class="btn genre-button cl-white bg-red">SEARCH</Button>
                </div>
            </div>
        );
    }
};