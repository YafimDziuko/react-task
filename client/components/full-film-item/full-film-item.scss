.full-film-item {
    display: flex;
    flex-direction: row;
    font-weight: 400;
    font-size: 16px;
    color: white;

    img {
        width: 300px;
        height: 350px;
    }

    .full-film-info {
        flex: 1;
        background-color: black;
        padding-left: 50px;

         .title-and-rating {
                display: flex;
                flex-direction: row;
                align-items: center;
                padding-top: 50px;
                margin-bottom: 10px;
                

            .title {
                color: #e64242;
                font-size: 30px;
                margin-right: 15px;
                font-weight: 700;
            }

            .rating {
                padding: 5px;
                border: 1px solid #fbff00;
                border-radius: 50%;
                color: #fbff00;
                font-size: 18px;
            }
        }

        .year-and-time {
            margin-top: 10px;
            margin-bottom: 20px;
            font-weight: 700;
        }

        .btn {
            font-weight: 700;
        }
    }
}