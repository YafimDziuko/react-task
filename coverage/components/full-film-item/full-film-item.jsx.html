<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/full-film-item/full-film-item.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/full-film-item</a> full-film-item.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/8</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
&nbsp;
import { fullLoad, returnToMainPage } from '../../store/action-creators';
&nbsp;
import './full-film-item.scss';
&nbsp;
class FullFilmItem extends React.Component {
<span class="fstat-no" title="function not covered" >    co</span>mponentWillMount() {
<span class="cstat-no" title="statement not covered" >        this.props.fullLoad(this.props.match.params.id);</span>
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder() {
<span class="cstat-no" title="statement not covered" >        if (this.props.user.fullItem.filmData) {</span>
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;div&gt;
            &lt;Link to='/' onClick={<span class="fstat-no" title="function not covered" >()</span>=&gt;{<span class="cstat-no" title="statement not covered" >console.log(this.props)}</span>}&gt;&lt;button className="btn return-button cl-red bg-white"&gt;SEARCH&lt;/button&gt;&lt;/Link&gt;
            &lt;div className="full-film-item"&gt;
                &lt;img src={this.props.user.fullItem.filmData.poster_path} /&gt;
                &lt;div className="full-film-info"&gt;
                    &lt;div className="title-and-rating"&gt;
                        &lt;p className="title"&gt;{this.props.user.fullItem.filmData.title}&lt;/p&gt;
                        { !!this.props.user.fullItem.filmData.vote_average &amp;&amp; 
                            &lt;p className="rating"&gt;{this.props.user.fullItem.filmData.vote_average}&lt;/p&gt;
                        }
                    &lt;/div&gt;
                    &lt;p&gt;{this.props.user.fullItem.filmData.tagline}&lt;/p&gt;
                    &lt;p className="year-and-time"&gt;
                        &lt;span className="year"&gt;{this.props.user.fullItem.filmData.release_date.slice(0, 4)}&lt;/span&gt;
                        { this.props.user.fullItem.filmData.runtime &amp;&amp;
                            &lt;span&gt;{this.props.user.fullItem.filmData.runtime} min&lt;/span&gt;
                        }
                    &lt;/p&gt;
                    &lt;p className="description"&gt;{this.props.user.fullItem.filmData.overview}&lt;/p&gt;
                &lt;/div&gt;
            &lt;/div&gt;
        &lt;/div&gt;
        );
    } else <span class="cstat-no" title="statement not covered" >return null;</span>
    }
};
&nbsp;
const mapDispatchToProps = {
    fullLoad
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        user: state,
    };
};
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(FullFilmItem);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun May 27 2018 19:49:32 GMT+0300 (MSK)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
