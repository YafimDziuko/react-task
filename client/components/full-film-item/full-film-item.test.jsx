import React from 'react';
import { FullFilmItem } from './full-film-item';

describe('<FullFilmItem/>', function() {
  it('should render FullFilmItem and match snapshot', function() {
      const wrapper = shallow(<FullFilmItem />);
      expect(wrapper).toMatchSnapshot();
  });
});