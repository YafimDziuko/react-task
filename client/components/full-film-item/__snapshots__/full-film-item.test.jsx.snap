// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullFilmItem/> should render FullFilmItem and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FullFilmItem
    additional="Oscar movie"
    description="A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film."
    duration={124}
    genre="horror"
    posterUrl="http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg"
    rating={4.7}
    title="Wolf wood"
    year={1992}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg"
        />,
        <div
          className="full-film-info"
        >
          <div
            className="title-and-rating"
          >
            <p
              className="title"
            >
              Wolf wood
            </p>
            <p
              className="rating"
            >
              4.7
            </p>
          </div>
          <p>
            Oscar movie
          </p>
          <p
            className="year-and-time"
          >
            1992
                
            124
             min
          </p>
          <p
            className="description"
          >
            A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.
          </p>
        </div>,
      ],
      "className": "full-film-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title-and-rating"
            >
              <p
                className="title"
              >
                Wolf wood
              </p>
              <p
                className="rating"
              >
                4.7
              </p>
            </div>,
            <p>
              Oscar movie
            </p>,
            <p
              className="year-and-time"
            >
              1992
                  
              124
               min
            </p>,
            <p
              className="description"
            >
              A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.
            </p>,
          ],
          "className": "full-film-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="title"
                >
                  Wolf wood
                </p>,
                <p
                  className="rating"
                >
                  4.7
                </p>,
              ],
              "className": "title-and-rating",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wolf wood",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Wolf wood",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 4.7,
                  "className": "rating",
                },
                "ref": null,
                "rendered": 4.7,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oscar movie",
            },
            "ref": null,
            "rendered": "Oscar movie",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                1992,
                "    ",
                124,
                " min",
              ],
              "className": "year-and-time",
            },
            "ref": null,
            "rendered": Array [
              1992,
              "    ",
              124,
              " min",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.",
              "className": "description",
            },
            "ref": null,
            "rendered": "A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg"
          />,
          <div
            className="full-film-info"
          >
            <div
              className="title-and-rating"
            >
              <p
                className="title"
              >
                Wolf wood
              </p>
              <p
                className="rating"
              >
                4.7
              </p>
            </div>
            <p>
              Oscar movie
            </p>
            <p
              className="year-and-time"
            >
              1992
                  
              124
               min
            </p>
            <p
              className="description"
            >
              A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.
            </p>
          </div>,
        ],
        "className": "full-film-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-and-rating"
              >
                <p
                  className="title"
                >
                  Wolf wood
                </p>
                <p
                  className="rating"
                >
                  4.7
                </p>
              </div>,
              <p>
                Oscar movie
              </p>,
              <p
                className="year-and-time"
              >
                1992
                    
                124
                 min
              </p>,
              <p
                className="description"
              >
                A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.
              </p>,
            ],
            "className": "full-film-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title"
                  >
                    Wolf wood
                  </p>,
                  <p
                    className="rating"
                  >
                    4.7
                  </p>,
                ],
                "className": "title-and-rating",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wolf wood",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Wolf wood",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 4.7,
                    "className": "rating",
                  },
                  "ref": null,
                  "rendered": 4.7,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Oscar movie",
              },
              "ref": null,
              "rendered": "Oscar movie",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  1992,
                  "    ",
                  124,
                  " min",
                ],
                "className": "year-and-time",
              },
              "ref": null,
              "rendered": Array [
                1992,
                "    ",
                124,
                " min",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.",
                "className": "description",
              },
              "ref": null,
              "rendered": "A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
